diff --git a/lib/termcap/tparam.c b/lib/termcap/tparam.c
index 7ef112b3..b6df1f8d 100644
--- a/lib/termcap/tparam.c
+++ b/lib/termcap/tparam.c
@@ -22,9 +22,9 @@ extern char *malloc ();
 extern char *realloc ();
 #endif
 
-#ifdef HAVE_UNISTD_H
-#include <unistd.h>
-#endif
+#if defined (HAVE_UNISTD_H) || defined (__linux__)
+#  include <unistd.h>
+#endif
 
 #if defined (HAVE_STRING_H)
 #include <string.h>
