diff --git a/lib/termcap/tparam.c b/lib/termcap/tparam.c
index c4bff088e73c5e382935398a23c3fcaf568e81c2..88bcb3056aa4b8f0843b0c7c87fa7ccbdd8e26da 100644
--- a/lib/termcap/tparam.c
+++ b/lib/termcap/tparam.c
@@ -30,6 +30,10 @@ extern char *realloc ();
 #endif
 
+#if defined (HAVE_UNISTD_H) || defined (__linux__)
+#  include <unistd.h>
+#endif
+
 #if defined (HAVE_STRING_H)
 #include <string.h>
 #endif
