--- a/meson.build	2024-02-27 17:26:04.000000000 +0000
+++ b/meson.build	2025-09-17 04:46:52.476708239 +0000
@@ -1013,9 +1013,11 @@
 libm = cc.find_library('m')
 libdl = cc.find_library('dl')
 libcrypt = dependency('libcrypt', required : false)
-if not libcrypt.found()
+if not libcrypt.found()
+        libcrypt = dependency('libxcrypt', required : false)
+endif
+libcrypt_found = libcrypt.found()
+if not libcrypt_found
         # fallback to use find_library() if libcrypt is provided by glibc, e.g. for LibreELEC.
         libcrypt = cc.find_library('crypt', required : false)
         libcrypt_found = libcrypt.found()
 endif
 libcap = dependency('libcap')
 
--- a/src/shared/meson.build	2024-02-27 17:26:04.000000000 +0000
+++ b/src/shared/meson.build	2025-09-17 04:46:54.824717720 +0000
@@ -316,7 +316,6 @@
                   libacl,
                   libblkid,
                   libcap,
-                  libcrypt,
                   libdl,
                   libgcrypt,
                   libiptc_cflags,
@@ -333,6 +332,10 @@
                   libxz,
                   libzstd]
 
+if libcrypt_found
+        libshared_deps += [libcrypt]
+endif
+
 libshared_sym_path = meson.current_source_dir() / 'libshared.sym'
 libshared_build_dir = meson.current_build_dir()
 
--- a/src/test/meson.build      2024-02-27 17:26:04.000000000 +0000
+++ b/src/test/meson.build      2025-09-17 04:47:20.520826070 +0000
@@ -196,0 +197,10 @@
+if libcrypt_found
+        executables += [
+                test_template + {
+                        'sources' : files('test-libcrypt-util.c'),
+                        'dependencies' : libcrypt,
+                        'timeout' : 120,
+                },
+        ]
+endif
+
@@ -305,5 +314,0 @@
-        test_template + {
-                'sources' : files('test-libcrypt-util.c'),
-                'dependencies' : libcrypt,
-                'timeout' : 120,
-        },
