--- a/meson_options.txt	2024-02-27 17:26:04.000000000 +0000
+++ b/meson_options.txt	2025-09-17 19:22:15.176444716 +0000
@@ -120,6 +120,8 @@
        description : 'install the systemd-localed stack')
 option('machined', type : 'boolean',
        description : 'install the systemd-machined stack')
+option('udev', type : 'boolean', value : true,
+       description : 'install the systemd-udevd stack')
 option('portabled', type : 'boolean',
        description : 'install the systemd-portabled stack')
 option('sysext', type : 'boolean',
@@ -152,6 +154,8 @@
        description : 'support for firstboot mechanism')
 option('randomseed', type : 'boolean',
        description : 'support for restoring random seed')
+option('removable', type : 'boolean', value : true,
+       description : 'install the systemd-remount-fs helper')
 option('backlight', type : 'boolean',
        description : 'support for restoring backlight state')
 option('vconsole', type : 'boolean',
--- a/meson.build	2024-02-27 17:26:04.000000000 +0000
+++ b/meson.build	2025-09-17 19:23:03.306870968 +0000
@@ -1603,6 +1603,7 @@
                 'pstore',
                 'quotacheck',
                 'randomseed',
+                'removable',
                 'resolve',
                 'rfkill',
                 'smack',
@@ -1612,6 +1613,7 @@
                 'timesyncd',
                 'tmpfiles',
                 'tpm',
+                'udev',
                 'userdb',
                 'utmp',
                 'vconsole',
@@ -2126,7 +2128,13 @@
 # systemd-networkd requires 'libsystemd_network'
 subdir('src/libsystemd-network')
 # hwdb requires 'udev_link_with' and 'udev_rpath'
-subdir('src/udev')
+if get_option('udev')
+	subdir('src/udev')
+else
+	udev_link_with = []
+	udev_rpath = ''
+	udev_pc = []
+endif
 
 subdir('src/ac-power')
 subdir('src/analyze')
@@ -2233,7 +2241,11 @@
 subdir('src/ukify/test')  # needs to be last for test_env variable
 subdir('test/fuzz')
 
-alias_target('devel', libsystemd_pc, libudev_pc, systemd_pc, udev_pc)
+devel_targets = [libsystemd_pc, libudev_pc, systemd_pc]
+if get_option('udev')
+	devel_targets += [udev_pc]
+endif
+alias_target('devel', devel_targets)
 
 ############################################################
 
@@ -2457,7 +2469,9 @@
 
 ############################################################
 
-subdir('rules.d')
+if get_option('udev')
+	subdir('rules.d')
+endif
 subdir('test')
 
 ############################################################
@@ -2801,6 +2815,7 @@
         ['pstore'],
         ['quotacheck'],
         ['randomseed'],
+        ['removable'],
         ['repart'],
         ['resolve'],
         ['rfkill'],
@@ -2812,6 +2827,7 @@
         ['timedated'],
         ['timesyncd'],
         ['tmpfiles'],
+        ['udev'],
         ['userdb'],
         ['vconsole'],
         ['vmspawn'],
--- a/src/remount-fs/meson.build	2024-02-27 17:26:04.000000000 +0000
+++ b/src/remount-fs/meson.build	2025-09-17 19:23:22.859853013 +0000
@@ -3,6 +3,7 @@
 executables += [
         libexec_template + {
                 'name' : 'systemd-remount-fs',
+                'conditions' : ['ENABLE_REMOVABLE'],
                 'sources' : files('remount-fs.c'),
         },
 ]
--- a/units/meson.build	2024-02-27 17:26:04.000000000 +0000
+++ b/units/meson.build	2025-09-17 19:23:40.388734993 +0000
@@ -95,7 +95,7 @@
         },
         {
           'file' : 'initrd-udevadm-cleanup-db.service',
-          'conditions' : ['ENABLE_INITRD'],
+          'conditions' : ['ENABLE_INITRD', 'ENABLE_UDEV'],
         },
         {
           'file' : 'initrd-usr-fs.target',
@@ -525,7 +525,10 @@
           'symlinks' : ['sysinit.target.wants/'],
         },
         { 'file' : 'systemd-reboot.service' },
-        { 'file' : 'systemd-remount-fs.service.in' },
+        {
+          'file' : 'systemd-remount-fs.service.in',
+          'conditions' : ['ENABLE_REMOVABLE'],
+        },
         {
           'file' : 'systemd-repart.service.in',
           'conditions' : ['ENABLE_REPART'],
@@ -632,21 +635,28 @@
           'conditions' : ['ENABLE_TMPFILES'],
           'symlinks' : ['sysinit.target.wants/'],
         },
-        { 'file' : 'systemd-udev-settle.service' },
+        {
+          'file' : 'systemd-udev-settle.service',
+          'conditions' : ['ENABLE_UDEV'],
+        },
         {
           'file' : 'systemd-udev-trigger.service',
+          'conditions' : ['ENABLE_UDEV'],
           'symlinks' : ['sysinit.target.wants/'],
         },
         {
           'file' : 'systemd-udevd-control.socket',
+          'conditions' : ['ENABLE_UDEV'],
           'symlinks' : ['sockets.target.wants/'],
         },
         {
           'file' : 'systemd-udevd-kernel.socket',
+          'conditions' : ['ENABLE_UDEV'],
           'symlinks' : ['sockets.target.wants/'],
         },
         {
           'file' : 'systemd-udevd.service.in',
+          'conditions' : ['ENABLE_UDEV'],
           'symlinks' : ['sysinit.target.wants/'],
         },
         {
